pkg/controllers/controller.go

```diff
@@ -59,16 +59,15 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 		apis.NewLivingConditionSet(conditions.Active).Manage(conditionsAccessor).MarkTrue(conditions.Active)
 	}
 	// 5. Reconcile
-	if _, err := c.Controller.Reconcile(ctx, resource); err != nil {
-		zap.S().Errorf("Controller failed to reconcile kind %s, %s",
-			resource.GetObjectKind().GroupVersionKind().Kind, err.Error())
-		return reconcile.Result{Requeue: true}, nil
+	_, err := c.Controller.Reconcile(ctx, resource)
+	if err != nil {
+		zap.S().Errorf("Controller failed to reconcile kind %s, %s", resource.GetObjectKind().GroupVersionKind().Kind, err.Error())
```

Comment:
So now we won't requeue right away once we hit an error? I am not sure if this can cause issues, and we might have missed testing this?
Why do we need to make this change though?

Reply:
After we fail a reconcile loop, we need to patch the status to say that the reconciliation failed. Regardless of if that patch fails or not, we will end up requeue-ing because returning an error requeues.